src/Template/Snippet.php

```diff
@@ -153,10 +153,14 @@ public function endslot(): void
 	 * or the template string for self-enclosed snippets
 	 */
 	public static function factory(
-		string|array $name,
+		string|array|null $name,
 		array $data = [],
 		bool $slots = false
 	): static|string {
+		if ($name === null) {
+			return '';
```

Comment:
Won't this break when using this:

```php
snippet(null, slots: true);
echo 'Some text';
endsnippet();
```

I think we need to let the rest of the code run and rather change the following line:

```php
$file = $name !== null ? static::file($name) : null;
```

Could you please also add a unit test where a single snippet name is passed as `null` but in slots mode?

Reply:
Yes, I didn't do it on purpose. Why are we running `slots` for a non-existent snippet? Is such a use correct or reasonable?

````php
snippet(null, slots: true);
echo 'Some text';
endsnippet();
````

What should the output of this be? Just `Some text`?

Reply:
I'd say the slot should be swallowed in this case as it's not intended to be printed directly. It's the same as if the slot is passed to an existing snippet but not used there (e.g. if the snippet is completely empty). So if we would print the slot for non-existing snippets, the behavior would be inconsistent. And potentially we would have issues with nested snippets (the wrong snippet would be closed).

Reply:
> the wrong snippet would be closed

You're right. I missed that. I've pushed the changes.